test/Steeltoe.Management.EndpointCore.Test/Security/SecurityMiddlewareTest.cs

```diff
@@ -0,0 +1,109 @@
+
+namespace Steeltoe.Management.Endpoint.Security.Test
+{
+    public class SecurityMiddlewareTest : BaseTest
+    {
+       [Fact]
+        public async void SecurityMiddleWare_ReturnsOKWhenNotSensitive()
+        {
+            var builder = GetBuilder<Startup>(new Dictionary<string, string>()
+            {
+                ["management:endpoints:enabled"] = "true",
+                ["management:endpoints:info:sensitive"] = "false"
+            });
+
+            using (var server = new TestServer(builder))
+            {
+                var client = server.CreateClient();
+                var result = await client.GetAsync("http://localhost/info");
+                Assert.Equal(HttpStatusCode.OK, result.StatusCode);
+            }
+
+            builder = GetBuilder<SecureStartup>(new Dictionary<string, string>()
+            {
+                ["management:endpoints:enabled"] = "true",
+                ["management:endpoints:info:sensitive"] = "false"
+            });
+
+            using (var server = new TestServer(builder))
+            {
+                var client = server.CreateClient();
+                client.DefaultRequestHeaders.Add("X-Test-Header", "value");
+
+                var result = await client.GetAsync("http://localhost/info");
+                Assert.Equal(HttpStatusCode.OK, result.StatusCode);
+            }
+        }
+
+        [Fact]
+        public async void SecurityMiddleWare_ReturnsUnauthorizedWhenSensitive()
+        {
+            var builder = GetBuilder<Startup>(new Dictionary<string, string>()
+            {
+                ["management:endpoints:enabled"] = "true",
+                ["management:endpoints:info:sensitive"] = "true"
+            });
+
+            using (var server = new TestServer(builder))
+            {
+                var client = server.CreateClient();
+                var result = await client.GetAsync("http://localhost/info");
+                Assert.Equal(HttpStatusCode.Unauthorized, result.StatusCode);
+            }
+        }
+
+        [Fact]
+        public async void SecurityMiddleWare_ReturnsOkWhenSensitive()
+        {
+            var builder = GetBuilder<SecureStartup>(new Dictionary<string, string>()
+            {
+                ["management:endpoints:enabled"] = "true",
+                ["management:endpoints:info:sensitive"] = "true",
+                ["management:endpoints:sensitiveclaim:type"] = "scope",
+                ["management:endpoints:sensitiveclaim:value"] = "actuator.read",
+            });
+
+            using (var server = new TestServer(builder))
+            {
+                var client = server.CreateClient();
+                client.DefaultRequestHeaders.Add("X-Test-Header", "value");
+                var result = await client.GetAsync("http://localhost/info");
+                Assert.Equal(HttpStatusCode.OK, result.StatusCode);
+            }
+        }
+
+       private IWebHostBuilder GetBuilder<T>(Dictionary<string, string> settings)
```

Comment:
I like this little helper, it might be a good idea to use something like this a little more broadly if we can